onnxruntime/core/providers/cuda/nn/conv.h

```diff
@@ -125,16 +126,62 @@ struct CudnnConvState {
   CudnnTensor y_tensor;
   CudnnConvolutionDescriptor conv_desc;
 
+  /*
+   Warning: Not thread-safe. Synchronization must be taken care of by the client code.
+   Will return cached memory if the requested size is the same as the cached memory's size
+  */
+  void* AllocMemoryForCuDnnConvResults(AllocatorPtr allocator, size_t size_in_bytes) {
```

Comment:
This is to allocate and hold onto the allocated memory to be re-used across several Run(). No need to keep re-allocating and freeing. It doesn't matter if this isn't thread-safe for now. The entire Conv Compute() is essentially executed only one thread at a time and hence having this method is safe.

Reply:
Probably can adopt a similar strategy for `workspace` buffer as well. Currently, we allocate and free this on every Run() in Conv and ConvTranspose. Probably, it can be cached. But, if using the arena, the allocation cost on every Run() may not be be too bad (not sure)